src/seedsigner/hardware/framebuffer.py

```diff
@@ -0,0 +1,40 @@
+from PIL import Image
+import mmap
+from seedsigner.hardware.RGBtoBGR import rgbtobgr
+
+class Framebuffer(object):
+    def __init__(self, device_no: int):
+        self.path = "/dev/fb%d" % device_no
+        config_dir = "/sys/class/graphics/fb%d/" % device_no
+        self.size = tuple(_read_config(config_dir + "virtual_size"))
+        self.width = self.size[0]
+        self.height = self.size[1]
+        self.stride = _read_config(config_dir + "stride")[0]
+        self.bits_per_pixel = _read_config(config_dir + "bits_per_pixel")[0]
+        assert self.stride == self.bits_per_pixel // 8 * self.size[0]
+        fp = open(self.path, "wb+")
+        self.length = self.size[0] * self.size[1] * (self.bits_per_pixel >> 3)
+        self.fb = mmap.mmap(fp.fileno(), length=self.length, access=mmap.ACCESS_WRITE)
+        print(self)
+
+    def __str__(self):
+        args = (self.path, self.size, self.stride, self.bits_per_pixel)
+        return "%s  size:%s  stride:%s  bits_per_pixel:%s" % args
+
+    def show(self, image: Image):
+        assert image.size == self.size
```

Comment:
We've generally opted to raise custom `Exception`s with more descriptive error messages where possible. Something like:

```
class FramebufferException(Exception):
    pass

....raise FramebufferException(f"Image size ({image.size}) does not equal framebuffer size ({self.size})")
```